src/plone/api/content.py

```diff
@@ -523,27 +523,30 @@ def get_view(name=None, context=None, request=None):
     # available, because the __init__ of said view will contain
     # errors in client code.
 
-    # Get all available views...
-    sm = getSiteManager()
-    available_views = sm.adapters.lookupAll(
-        required=(providedBy(context), providedBy(request)),
-        provided=Interface,
-    )
-    # and get their names.
-    available_view_names = [view[0] for view in available_views]
-
-    # Raise an error if the requested view is not available.
-    if name not in available_view_names:
-        raise InvalidParameterError(
-            "Cannot find a view with name '{name}'.\n"
-            "Available views are:\n"
-            "{views}".format(
-                name=name,
-                views="\n".join(sorted(available_view_names)),
-            ),
+    try:
+        adapter = getMultiAdapter((context, request), name=name)
+    except Exception: 
+
+        #Getting all available views
+        sm = getSiteManager()
+        available_views = sm.adapters.lookupAll(
+            required=(providedBy(context), providedBy(request)),
+            provided=Interface,
         )
-    return getMultiAdapter((context, request), name=name)
-
+        
+        # Check if the requested view is available by getting the names of all available views
+        available_view_names = [view[0] for view in available_views]
+        if name not in available_view_names:
+            # Raise an error if the requested view is not available.
+            raise InvalidParameterError(
+                "Cannot find a view with name '{name}'.\n"
+                "Available views are:\n"
+                "{views}".format(
+                    name=name,
+                    views="\n".join(sorted(available_view_names)),
+                ),
+            )
+    return adapter
```

Comment:
`adapter` here could be not defined, if we get an exception when trying to get the adapter, and thus get into the `exception Exception` part of the code and the view name it is found in the `available_view_names`.

Could that happen, though? 🤔 A view that you don't have permissions for would get you in such corner case? 🤔

Reply:
The security is not checked when you call `getMultiAdapter`, anyway the `except Exception` is not good IMO.